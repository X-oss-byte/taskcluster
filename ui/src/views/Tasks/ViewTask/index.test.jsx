import React from 'react';
import { render, waitFor, act } from '@testing-library/react';
import { ApolloProvider } from 'react-apollo';
import setupClient from 'apollo-client-mock';
import { MemoryRouter } from 'react-router-dom';
import ViewTask from './index';

const typeDefs = `
  type Task {
    task: Task!
  }
`;
const defaultMocks = {
  Task: () => ({
    task: () => ({
      taskId: 'taskId',
    }),
  }),
};

describe('ViewTask page', () => {
  it('should render ViewTask page', async () => {
    const createClient = setupClient(defaultMocks, typeDefs);

    await act(async () => {
      const { asFragment } = render(
        <MemoryRouter keyLength={0}>
          <ApolloProvider client={createClient()}>
            <ViewTask match={{ params: {} }} task={{ taskId: 'taskId' }} />
          </ApolloProvider>
        </MemoryRouter>
      );

      await waitFor(() => {});

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
