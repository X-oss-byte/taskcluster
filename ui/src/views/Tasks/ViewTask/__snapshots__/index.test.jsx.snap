// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewTask page should render ViewTask page 1`] = `<DocumentFragment />`;
